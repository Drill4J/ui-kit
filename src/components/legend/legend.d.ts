/// <reference types="react" />
interface LegendItem {
  color: string;
  borderColor?: string;
  label: React.ReactNode;
}
interface Props {
  className?: string;
  legendItems: LegendItem[];
}
export declare const Legend: {
  ({ className, legendItems }: Props): JSX.Element;
  displayName: string;
  __docgenInfo: {
    description: string;
    displayName: string;
    props: {
      className: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      legendItems: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
    };
  };
};
export {};
