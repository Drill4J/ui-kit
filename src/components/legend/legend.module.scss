@import '../../style-guide/colors';

.legend {
  display: flex;
  gap: 16px;
  font-size: 12px;
  line-height: 16px;
  color: map-get($monochrome, default);

  &__circle {
    margin-right: 8px;
  }
}
