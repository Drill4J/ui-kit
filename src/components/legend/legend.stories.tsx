import { storiesOf } from '@storybook/react';

import { Legend } from './legend';

storiesOf('Legend', module).add('visualisation colors', () => (
  <Legend legendItems={[
    { label: 'no fill', color: '#fff', borderColor: '#00b602' },
    { label: 'overlapping', color: '#4f8ac9' },
    { label: 'scope-cover', color: '#aed4fd' },
  ]}
  />
));
