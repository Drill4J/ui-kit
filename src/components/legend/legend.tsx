import 'twin.macro';

export interface LegendItem {
  color: string;
  borderColor?: string;
  label: React.ReactNode;
}

interface Props {
  legendItems: LegendItem[];
}

export const Legend = ({ legendItems }: Props) => (
  <div tw="flex gap-x-6 text-12 leading-16 text-monochrome-default">
    {legendItems.map(({ color, label, borderColor }) => (
      <div tw="flex items-center gap-x-2" key={color}>
        <svg width={10} height={10}>
          <circle cx="5" cy="5" r="4.5" fill={color} stroke={borderColor} data-test={`legend:circle:${label}`} />
        </svg>
        {label}
      </div>
    ))}
  </div>
);
