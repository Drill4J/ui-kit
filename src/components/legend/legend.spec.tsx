import renderer from 'react-test-renderer';

import { Legend } from './legend';

describe('Legend', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Legend legendItems={[
      { label: 'build-cover', color: '#62adfc' },
      { label: 'overlapping', color: '#4f8ac9' },
      { label: 'scope-cover', color: '#aed4fd' },
    ]}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
