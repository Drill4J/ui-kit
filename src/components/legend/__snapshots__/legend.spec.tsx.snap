// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend should match snapshot 1`] = `
<div
  className="sc-bdvvaa htzGmK"
>
  <div
    className="sc-gsDJrp iCEvit"
  >
    <svg
      height={10}
      width={10}
    >
      <circle
        cx="5"
        cy="5"
        fill="#62adfc"
        r="4.5"
      />
    </svg>
    build-cover
  </div>
  <div
    className="sc-gsDJrp iCEvit"
  >
    <svg
      height={10}
      width={10}
    >
      <circle
        cx="5"
        cy="5"
        fill="#4f8ac9"
        r="4.5"
      />
    </svg>
    overlapping
  </div>
  <div
    className="sc-gsDJrp iCEvit"
  >
    <svg
      height={10}
      width={10}
    >
      <circle
        cx="5"
        cy="5"
        fill="#aed4fd"
        r="4.5"
      />
    </svg>
    scope-cover
  </div>
</div>
`;
